subprojects/core/src/integTest/groovy/org/gradle/api/internal/initialization/loadercache/ClassLoadersCachingIntegrationTest.groovy

```diff
@@ -16,26 +16,11 @@
 
 package org.gradle.api.internal.initialization.loadercache
 
-import org.gradle.integtests.fixtures.executer.ExecutionResult
+
 import org.gradle.integtests.fixtures.longlived.PersistentBuildProcessIntegrationTest
 
 class ClassLoadersCachingIntegrationTest extends PersistentBuildProcessIntegrationTest {
 
-    def cacheSizePerRun = []
```

Comment:
I've removed the cache size checking from this integration test, as this is now more complex. Classloaders will get cleaned up in a subsequent session, making the "size change" look very unintuitive based on just the file changes that the test does. 

The correct usage of the `remove` method is already unit-tested in other areas. I've added a soak test that proves that we eventually collect old build script classloaders.